tajo-core/src/main/java/org/apache/tajo/master/QueryInProgress.java

```diff
@@ -163,6 +168,12 @@ public void submmitQueryToMaster() {
 
     try {
       writeLock.lockInterruptibly();
+    } catch (Exception e) {
```

Comment:
It just excluded locking from try/finally clause, avoiding releasing lock which is not held by the thread.